src/main/webapp/app/components/tabs/curation-data-import-tab/import-methods.tsx

```diff
@@ -0,0 +1,329 @@
+        comment: '',
+        excluding: [],
+        genes: annotatedAlteration.genes || [],
+        proteinChange: dataRow.data.protein_change ?? annotatedAlteration.proteinChange,
+        proteinStart: annotatedAlteration.start,
+        proteinEnd: annotatedAlteration.end,
+        refResidues: annotatedAlteration.refResidues || '',
+        varResidues: annotatedAlteration.variantResidues || '',
+      },
+    ];
+  } catch (e) {
+    return {
+      status: 'error',
+      message: "We aren't able to annotate the alteration. Please reach out to dev team.",
+    };
+  }
+
+  const isGermlineData = (data: GermlineMutationDI | SomaticMutationDI): data is GermlineMutationDI => {
+    return isGermline;
+  };
+  if (isGermlineData(dataRow.data)) {
+    const data = dataRow.data;
+    if (data.pathogenicity) {
+      const importStatus: DataImportStatus = pathogenicityCheck(data.pathogenicity);
+      if (importStatus.status === 'error') {
+        return importStatus;
+      }
+    }
+    mutation.mutation_effect.pathogenic = data.pathogenicity as PATHOGENICITY;
+    mutation.mutation_effect.pathogenic_review = new Review(authStore.fullName, '');
+    mutationImpactStatusUpdated = true;
+  } else {
+    const data = dataRow.data;
+    if (data.oncogenicity) {
+      const importStatus: DataImportStatus = oncogenicityCheck(data.oncogenicity);
+      if (importStatus.status === 'error') {
+        return importStatus;
+      }
+    }
+    mutation.mutation_effect.oncogenic = data.oncogenicity as FIREBASE_ONCOGENICITY;
+    mutation.mutation_effect.oncogenic_review = new Review(authStore.fullName, '');
+    mutationImpactStatusUpdated = true;
+  }
+
+  const existingMuts = getDuplicateMutations([mutation.name], mutationList, vusList, {
+    useFullAlterationName: true,
+    exact: true,
+  });
+
+  // Delete existing VUS before importing
+  for (const mut of existingMuts) {
+    if (mut.inVusList) {
+      await firebaseGeneService.firebaseRepository.delete(`${getFirebaseVusPath(isGermline, hugoSymbol)}/${mut.duplicate}`);
+    } else {
+      // if alteration exists in the Mutation list, we return and give error
+      return {
+        status: 'error',
+        message: 'Mutation exists in gene, we cannot overwrite automatically. Please update manually.',
+      };
+    }
+  }
+
+  try {
+    await firebaseGeneService
+      .addMutation(`${getFirebaseGenePath(isGermline, hugoSymbol)}/mutations`, mutation, isGermline, false, dataRow.data.description)
+      .then(async () => {
+        if (mutationImpactStatusUpdated) {
+          let uuid: string;
+          if (isGermline) {
+            uuid = mutation.mutation_effect.pathogenic_uuid;
+          } else {
+            uuid = mutation.mutation_effect.oncogenic_uuid;
+          }
+          // I can't use updateReviewableContent here due to lacking of the firebase path
+          await firebaseMetaService.updateGeneReviewUuid(hugoSymbol, uuid, true, isGermline);
```

Comment:
I would suggest using `firebaseMetaService.updateMeta`, which will add the uuid to review and also update the last edited information.